### PR TITLE
[Issue #20] Add static methods for operations without having to initi…

### PR DESCRIPTION
**Pull request by GopinathRakkiyannan**
I have created static methods for the below instance methods

- AddHours
- AddMinutes
- AddSeconds
- AddFrames

I have not created static method for **ConvertFramerate** as it is not required as per my analysis.

Please review them and let me know your inputs. If all are good, Please approve my PR.

### CODE REVIEW

**Comment by GopinathRakkiyannan** [comment]
@RasmusBroborg 

Good day, As I am not much familiar in github, Could you please help me to resolve the conflicts. Thank you in advance

**Comment by RasmusBroborg** [comment]
Hi @GopinathRakkiyannan,

Thank you for submitting the pull request. I will review the code shortly, and make sure to submit code suggestions for you to add to fix any potential merge conflict and the like. Once the review has been applied, simply add the suggestions to your code, make a new commit, and the changes will update the pull request.

File: src/timecode/Timecode.cs [hunk 1]
```diff
@@ -490,6 +491,73 @@ private void SetTotalFramesUsingDropFrames(int hours, int minutes, int seconds,
       int totalMinutes = (60 * hours) + minutes;
       int totalFrames = ((hourFrames * hours) + (minuteFrames * minutes) + (timeBase * seconds) + frames) - (dropFrames * (totalMinutes - (totalMinutes / 10)));
       TotalFrames = totalFrames;
+    }
+
+    #region staticMethods
+
```
Comment (RasmusBroborg) [inline src/timecode/Timecode.cs:479]: These additions are great. We do need to implement a static call for the method ConvertFramerate() (such as the provided initial example in issue #20). I will assist you by writing a suggestion which you can add directly using your Github, or by updating your code and push a new commit. See below for the suggestion:

```suggestion
/// <summary>
    /// Converts a timecode string to a timecode string of a different framerate.
    /// </summary>
    /// <param name="originalTimecode">The original timecode, formatted "HH:MM:SS:FF" or "-HH:MM:SS:FF".</param>
    /// <param name="originalFramerate">The original timecode framerate.</param>
    /// <param name="destinationFramerate">The target framerate to convert to.</param>
    /// <returns></returns>
    /// <exception cref="ArgumentException"></exception>
    public static string ConvertFramerate(string originalTimecode, Framerate originalFramerate, Framerate destinationFramerate)
    {
      // Validate the original timecode format
      Regex tcRegex = new Regex(RegexPattern);
      if (!tcRegex.IsMatch(originalTimecode))
        throw new ArgumentException("Invalid timecode format.", nameof(originalTimecode));

      Timecode timecode = new Timecode(originalTimecode, originalFramerate);
      timecode.ConvertFramerate(destinationFramerate);
      return timecode.ToString();
    }
```

File: tests/timecode.test/Timecode_Test.cs [hunk 2]
```diff
@@ -377,6 +377,94 @@ public void MoreThan_Timecodes_With_Different_Framerates()
 
       // Assert
       comparison.Should().Throw<InvalidOperationException>();
+    }
+
+    #region staticmethodtests
+
```
Comment (RasmusBroborg) [inline tests/timecode.test/Timecode_Test.cs:390]: As we will implement a new static method we should add tests for it as well. I will suggest tests below:

```suggestion
[Fact]
    public void Convert_FrameRate_Using_Static_Method()
    {
      var sut = Timecode.ConvertFramerate("10:00:00:00", Enums.Framerate.fps23_976, Enums.Framerate.fps59_94_NDF);
      sut.Should().Be("04:00:00:00");
    }

```

**Review by RasmusBroborg — CHANGES_REQUESTED** [review]
Good job!

I have submittet comments with code suggestions which need to be updated. Once the comments have been resolved I will approve the PR and begin the automated test workflow.

File: src/timecode/Timecode.cs (hunk 1 above)
Comment (GopinathRakkiyannan) [reply src/timecode/Timecode.cs]: @RasmusBroborg Above review comments applied and committed the changes

File: tests/timecode.test/Timecode_Test.cs (hunk 2 above)
Comment (GopinathRakkiyannan) [reply tests/timecode.test/Timecode_Test.cs:390]: @RasmusBroborg Above review comments applied and committed the changes

**Comment by RasmusBroborg** [comment]
@GopinathRakkiyannan Great, you have applied your changes and all tests have passed. We are almost ready to approve and merge.

If we look at the pull request it now appears thar the files have accidentally changed our line termination format, which is why it now says that +1,121 rows have been added and −939 rows have been deleted. This could be a bug introduced by Github, as the format should be handled by the editorconfig-file. To fix this, we need to save the files locally to trigger the editorconfig's autoformatter. 

Do the following steps:

- Go to your local project
- `git pull`
- Open the file Timecode.cs and save it.
- Open the file Timecode_Test.cs  and save it.
- `git add -A`
- `git commit -m "Fix format changes"`
- `git push`

After this we should see that the files format is back to normal and we are ready to approve the PR.

**Comment by GopinathRakkiyannan** [comment]
> @GopinathRakkiyannan Great, you have applied your changes and all tests have passed. We are almost ready to approve and merge.
> 
> If we look at the pull request it now appears thar the files have accidentally changed our line termination format, which is why it now says that +1,121 rows have been added and −939 rows have been deleted. This could be a bug introduced by Github, as the format should be handled by the editorconfig file. To fix this, we need to save the files locally to trigger the editorconfig's autoformatter.
> 
> Do the following steps:
> 
> * Go to your local project
> * `git pull`
> * Open the file Timecode.cs and save it.
> * Open the file Timecode_Test.cs  and save it.
> * `git add -A`
> * `git commit -m "Fix format changes"`
> * `git push`
> 
> After this we should see that the files format is back to normal and we are ready to approve the PR.

@RasmusBroborg Could you please verify and provide your confirmation on the same

**Comment by RasmusBroborg** [comment]
@GopinathRakkiyannan I believe to have found the culprit. The project was missing a .gitattributes file, which I have now added to the main project.

Try adding this file and create a new commit: https://github.com/RasmusBroborg/dotnet-timecode/blob/main/.gitattributes

**Comment by leijon** [comment]
@GopinathRakkiyannan  @RasmusBroborg you can also update Gopinaths repository by clicking the button which says something like "2 commits behind of RasmusBroborg:main" to fetch the .gitattributes file. It should show up right next to the "4 commits ahead of RasmusBroborg:main". It could solve the issue.

**Comment by GopinathRakkiyannan** [comment]
@RasmusBroborg  I have updated my repository as you mentioned. Please check now

**Comment by RasmusBroborg** [comment]
@GopinathRakkiyannan There is no visual confirmation that the bug is fixed, but the .gitattributes should fix it when merging into main. I will accept the PR and merge with main and deal with any whitespace related issues if they show up.

Thanks for contributing!

**Review by RasmusBroborg — APPROVED** [review]

**Comment by GopinathRakkiyannan** [comment]
Thanks lot @RasmusBroborg  giving me a chance to work on it.!

**Pull request merged**